### PR TITLE
Add SubjectGroup selection api end points

### PR DESCRIPTION
**Pull request by camallen**
Linked to the work in #3521, #3511 and #3520 

Add GroupSubject selection API end points for clients to request SubjectGroups be created on the fly. This end point relies heavily on the existing selector services but instead of returning the normal Subject resource it will create a set of SubjectGroup resources from the returned list of selected subjects and serialize the page of `group_subject` Subject resources. These `group_subject` resources are representations of each SubjectGroup to fits our existing PFE code data model (keep talk working) as well as our new FEM code to display the Subject Groups for classification UX. 

the API end point is similar to the queued end point
```
GET /api/v1/subjects#queued
GET /api/subjects/grouped
```
It requires the following query params `?workflow_id=1,num_rows=1,num_columns=1` with the `subject_set_id` being an optional value as per the `queued` end point.

The end point sanity checks the `num_rows` and `num_columns` inputs are Integers between 1 and 10 to ensure we don't have large groups being created for decent UX. 

It also ensure the workflow has a relevant, matching config setup to drive this feature (both in the UI client and the API), consistency between these values is required.

All the created SubjectGroup.group_subjects will belong to the `SUBJECT_GROUP_UPLOADER_ID` set via the ENV var, by default this will be set this to the Zooniverse user.

Finally this feature is behind a simple allow list through the ENV `SUBJECT_GROUP_WORFKLOW_ID_ALLOWLIST` var which takes a list of allowed workflows that can use this feature, e.g. `SUBJECT_GROUP_WORFKLOW_ID_ALLOWLIST=1,2,3`. Without this being set the end point will raise with a `ApiErrors::FeatureDisabled` error.

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

File: kubernetes/deployment-staging.tmpl [hunk 1]
```diff
@@ -92,6 +92,8 @@ data:
   STORAGE_URL: https://panoptes-uploads-staging.zooniverse.org/
   STORAGE_BUCKET: zooniverse-static
   STORAGE_PREFIX: panoptes-uploads.zooniverse.org/staging/
+  SUBJECT_GROUP_UPLOADER_ID: '1'
+  SUBJECT_GROUP_WORFKLOW_ID_ALLOWLIST: '3451'
```
Comment (camallen) [inline kubernetes/deployment-staging.tmpl:96]:
```suggestion
  SUBJECT_GROUP_WORFKLOW_ID_ALLOWLIST: '3451,3402,3412'
```

**Review by nciemniak — COMMENTED** [review]
Looks good apart from the spelling for the one env var, though I do think it would be useful to document (somewhere even if just in the PR description) the exact workflow config keys/format we are expecting to see for this setup - or at least an example of the expected configuration format, like this:
```ruby
workflow.configuration = { 'subject_group': {'num_columns': 3, 'num_rows': 3 } }
```
Would be nice not to have to dig into tests to find this.
I'm going to do one more pass on this tomorrow because there are certain things that didnt click 100% in my brain today but I think we're good here.

One question for my curiosity - what is a scenario in which a user would need to create a subject group via this endpoint? Or is it an endpoint that we intend for PFE to use for a specific scenario?

File: kubernetes/deployment-staging.tmpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (camallen) [reply kubernetes/deployment-staging.tmpl]: addressed in https://github.com/zooniverse/panoptes/pull/3533

**Comment by camallen** [comment]
> One question for my curiosity - what is a scenario in which a user would need to create a subject group via this endpoint? Or is it an endpoint that we intend for PFE to use for a specific scenario?

This is for a specific project we're building with the Survos project team on the FEM codebase:
- https://github.com/zooniverse/front-end-monorepo/pull/1556
- https://github.com/zooniverse/front-end-monorepo/pull/1557

If it proves successful we can roll it out to general availability but for the shorter term this is an experimental feature only allowed for specific projects / workflows.

**Pull request merged**